### ISSUE TITLE
Local previewing in container is broken

### ISSUE THREAD

**Issue by sftim** [kubernetes/website#47673]
**This is a Bug Report**

**Problem:**
You cannot run `make container-serve` to get a local website preview, nor can you run `make container-image` to build the container locally.

**Proposed Solution:**
Fix local previewing by ensuring that `make container-image` builds a working container image

**Additional Information:**
This is relevant to contributor experience.

---

Optionally, we could run CI tests whenever there is a change to `hugo.toml`, `netlify.toml` or `Makefile`, and verify that we can:
- build a container image
- run Hugo inside that image (smoke test)

**Comment by mbianchidev** [comment]
@tengqm you can read more on this thread https://kubernetes.slack.com/archives/C1J0BPD2M/p1724428225643559?thread_ts=1724226266.705639&cid=C1J0BPD2M

TLDR; the error was:
`#10 2.275 /root/src/hugo-0.133.0/go.mod:172: invalid go version '1.22.6': must match format 1.23`

![image](https://github.com/user-attachments/assets/758e7337-95a0-4792-9343-3502744bcd75)

Due to the latest Hugo upgrade that requires Golang 1.23 when building the image.

The fix works locally:

![image](https://github.com/user-attachments/assets/29e5b35b-205e-4d76-9272-214797dd2279)

@sftim +1 to running a test in the CI.
Would that be something completely new or a pre-existing pipeline to enhance? Asking because I don't see any other GitHub workflow aside from `netlify-periodic-build.yml`.

_Note: I'm still getting accustomed to the codebase so this question might sound silly._

**Comment by tengqm** [comment]
Got it. A change to Go version is necessary.

**Comment by sftim** [comment]
> I am unable to produce this because being blocked to gcr.io, the command never worked for me (in China). I'm curious though what kind of errors we have encountered and why an upgrade of Go is required.

To test the container image build locally, run `make container-image`. You need access to the Docker registry; you don't need access to push to Google.
